### PR TITLE
test to catch illegal nodeNames returned by PreFilterResult

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind cleanup
/sig scheduling

#### What this PR does / why we need it:

A minimum test to reproduce #123465, but in a general way.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

Why the fix doesn't exist in this PR is b/c the fix exists as a by-product of #119779. Backporting-wise, we may manually carve out of PR w/ this test along with extracting a minimum fix from #119779

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
cc @alculquicondor @sanposhiho 

Backporting-wise, we may manually carve out of PR w/ this test along with extracting a minimum fix from https://github.com/kubernetes/kubernetes/pull/119779.

Once we have a minimum portable PR, for in-tree plugins, it's needed for [1.27, 1.29]; for out-of-tree plugins, [version PreFilterResult was introduced, 1.29].

**Review by sanposhiho — COMMENTED** [review]
Thanks, small nit only

File: test/integration/scheduler/plugins/plugins_test.go [hunk 1]
```diff
@@ -80,6 +81,7 @@ type PreFilterPlugin struct {
 	numPreFilterCalled int
 	failPreFilter      bool
 	rejectPreFilter    bool
+	injectNodeNames    sets.Set[string]
```
[the remark this answers is not in the document]
Comment (Huang-Wei) [reply test/integration/scheduler/plugins/plugins_test.go]: done.

**Review by sanposhiho — COMMENTED** [review]
/lgtm

**Pull request merged**